when BUILD_WITHOUT_CURL=ON and BUILD_MAVSDK_SERVER=OFF

After compile the mavsdk library with  BUILD_WITHOUT_CURL=ON and BUILD_MAVSDK_SERVER=OFF, and I build the offboard example, the following error occurs:
libmavsdk.so: undefined reference to `mavsdk::HttpLoader::~HttpLoader()'

libmavsdk.so: undefined reference to `mavsdk::HttpLoader::HttpLoader()'

libmavsdk.so: undefined reference to `mavsdk::HttpLoader::download_async(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<void (int, mavsdk::HttpStatus, CURLcode)>)'

It looks like this was broken on `main`:

```
/home/julianoes/src/upstream/MAVSDK/src/mavsdk/plugins/camera/camera_impl.cpp: In member function ‘void mavsdk::CameraImpl::check_camera_definition_with_lock(PotentialCamera&)’:
/home/julianoes/src/upstream/MAVSDK/src/mavsdk/plugins/camera/camera_impl.cpp:1310:13: error: ‘notify_camera_list’ was not declared in this scope
 1310 |             notify_camera_list();
```

Once I fixed that I could reproduce your issue:

```
/usr/bin/ld: src/mavsdk/libmavsdk.so.3.2.0: undefined reference to `mavsdk::HttpLoader::~HttpLoader()'
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -167,6 +167,27 @@ jobs:
       - name: test (mavsdk_server)
         run: ./build/release/src/mavsdk_server/test/unit_tests_mavsdk_server
 
+  ubuntu-no-curl:
+    name: Ubuntu 24.04 (mavsdk, without curl)
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - uses: actions/cache@v4
+        id: cache
+        with:
+          path: ./build/release/third_party/install
+          key: ${{ github.job }}-${{ hashFiles('./third_party/**') }}
+      - name: disable superbuild on cache hit
+        if: steps.cache.outputs.cache-hit == 'true'
+        run: echo "superbuild=-DSUPERBUILD=OFF" >> $GITHUB_ENV
+      - name: configure
+        run: |
+          cmake $superbuild -DCMAKE_PREFIX_PATH=$(pwd)/build/release/third_party/install -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_MAVSDK_SERVER=OFF -DBUILD_WITHOUT_CURL=ON -DWERROR=ON -DCMAKE_INSTALL_PREFIX=install -Bbuild/release -H.
+      - name: build
+        run: cmake --build build/release -j2
+
   ubuntu-hunter:
     name: ${{ matrix.ubuntu_image }} (non-mavsdk_server, hunter)
     runs-on: ${{ matrix.ubuntu_image }}

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -44,7 +44,9 @@ hunter_add_package(jsoncpp)
 hunter_add_package(CURL)
 hunter_add_package(lzma)
 
-find_package(CURL REQUIRED)
+if(NOT BUILD_WITHOUT_CURL)
+    find_package(CURL REQUIRED)
+endif()
 find_package(LibLZMA REQUIRED)
 
 if(BUILD_TESTING AND (IOS OR ANDROID))

--- a/src/mavsdk/core/mavlink_component_metadata.h
+++ b/src/mavsdk/core/mavlink_component_metadata.h
@@ -2,9 +2,12 @@
 
 #include "callback_list.h"
 #include "file_cache.h"
-#include "http_loader.h"
 #include "mavlink_command_sender.h"
 #include <json/json.h>
+
+#if BUILD_WITHOUT_CURL != 1
+#include "http_loader.h"
+#endif
 
 #include <optional>
 #include <set>

--- a/src/mavsdk/plugins/camera/camera_impl.cpp
+++ b/src/mavsdk/plugins/camera/camera_impl.cpp
@@ -1,12 +1,15 @@
 #include "camera_impl.h"
 #include "camera_definition.h"
 #include "system.h"
-#include "http_loader.h"
 #include "unused.h"
 #include "callback_list.tpp"
 #include "fs_utils.h"
 #include "string_utils.h"
 #include "math_utils.h"
+
+#if BUILD_WITHOUT_CURL != 1
+#include "http_loader.h"
+#endif
 
 #include <algorithm>
 #include <cassert>

--- a/src/mavsdk/plugins/camera/camera_impl.cpp
+++ b/src/mavsdk/plugins/camera/camera_impl.cpp
@@ -1307,7 +1307,7 @@ void CameraImpl::check_camera_definition_with_lock(PotentialCamera& potential_ca
         } else if (starts_with(url, "http://") || starts_with(url, "https://")) {
 #if BUILD_WITHOUT_CURL == 1
             potential_camera.camera_definition_result = Camera::Result::ProtocolUnsupported;
-            notify_camera_list();
+            notify_camera_list_with_lock();
 #else
             if (_http_loader == nullptr) {
                 _http_loader = std::make_unique<HttpLoader>();

--- a/src/mavsdk/plugins/camera/camera_impl.h
+++ b/src/mavsdk/plugins/camera/camera_impl.h
@@ -386,7 +386,9 @@ private:
     CallEveryHandler::Cookie _request_slower_call_every_cookie{};
     CallEveryHandler::Cookie _request_faster_call_every_cookie{};
 
+#if BUILD_WITHOUT_CURL != 1
     std::unique_ptr<HttpLoader> _http_loader{nullptr};
+#endif
 
     bool _debugging{false};
 };

--- a/src/unit_tests/CMakeLists.txt
+++ b/src/unit_tests/CMakeLists.txt
@@ -15,7 +15,6 @@ find_package(tinyxml2 REQUIRED)
 
 target_link_libraries(unit_tests_runner
     mavsdk
-    CURL::libcurl
     JsonCpp::JsonCpp
     tinyxml2::tinyxml2
     GTest::gtest
@@ -26,6 +25,11 @@ target_link_libraries(unit_tests_runner
 target_include_directories(unit_tests_runner
     PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../mavsdk/core
     PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/../mavsdk/core
+)
+
+target_compile_definitions(unit_tests_runner
+    PRIVATE
+    BUILD_WITHOUT_CURL=1
 )
 
 if (MSVC AND BUILD_SHARED_LIBS)

